src/request-filtering-agent.ts

```diff
@@ -39,23 +39,34 @@ export interface RequestFilteringAgentOptions {
  * @param family optional
  * @param options
  */
-const validateAddress = ({ address, host, family }: { address: string; host?: string; family?: string | number }, options: Required<RequestFilteringAgentOptions>) => {
-    // prefer allowed list
-    if (options.allowIPAddressList.length > 0 && options.allowIPAddressList.includes(address)) {
+const validateIPAddress = ({ address, host, family }: { address: string; host?: string; family?: string | number }, options: Required<RequestFilteringAgentOptions>) => {
+    // if it is not IP address, skip it
+    if (net.isIP(address) === 0) {
         return;
     }
-
-    if (!options.allowMetaIPAddress) {
-        if (address === "0.0.0.0" || address == "::") {
-            return new Error(`DNS lookup ${address}(family:${family}, host:${host}) is not allowed. Because, It is meta IP address.`);
+    try {
+        const addr = ipaddr.parse(address);
+        const range = addr.range();
+        // prefer allowed list
+        if (options.allowIPAddressList.length > 0 && options.allowIPAddressList.includes(address)) {
+            return;
+        }
+        if (!options.allowMetaIPAddress) {
+            // address === "0.0.0.0" || address == "::"
+            if (range === "unspecified") {
+                return new Error(`DNS lookup ${address}(family:${family}, host:${host}) is not allowed. Because, It is meta IP address.`);
+            }
+        }
+        // TODO: rename option name
+        if (!options.allowPrivateIPAddress && range !== "unicast") {
```

Comment:
I think that `allowPrivateIPAddress` option name is invalid.
`allowReservedIPAddress` is correct.